arcgis-ios-sdk-samples/Layers/Display WFS/DisplayWFSViewController.swift

```diff
@@ -0,0 +1,98 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class DisplayWFSViewController: UIViewController {
+
+    @IBOutlet private weak var mapView: AGSMapView!
+    private var wfsFeatureTable : AGSWFSFeatureTable!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Initialize a map with topographic basemap
+        let map = AGSMap(basemap: .topographic())
+        
+        // A URL to the GetCapabilities endpoint of a WFS service
+        let wfsServiceURL = URL(string: "https://dservices2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/services/Seattle_Downtown_Features/WFSServer?service=wfs&request=getcapabilities")!
+        
+        // Name of the table to load
+        let wfsTableName = "Seattle_Downtown_Features:Buildings"
+        
+        // Initialize a WFS feature table with service URL and uniquely identifying table name
+        wfsFeatureTable = AGSWFSFeatureTable(url: wfsServiceURL, tableName: wfsTableName)
+        
+        // Set feature request mode to manual - only manual is supported at v100.5.
+        // In this mode, you must manually populate the table - panning and zooming won't request features automatically.
+        wfsFeatureTable.featureRequestMode = .manualCache
+    
+        // Initialize a feature layer from WFS feature table
+        let wfsFeatureLayer = AGSFeatureLayer(featureTable: self.wfsFeatureTable)
+        
+        // Create a simple renderer
+        let simpleLineSymbol = AGSSimpleLineSymbol(style: .solid, color: .red, width: 3.0)
+        let renderer = AGSSimpleRenderer(symbol: simpleLineSymbol)
+        
+        // Apply renderer to feature layer
+        wfsFeatureLayer.renderer = renderer
+        
+        // Add feature layer to the map
+        map.operationalLayers.add(wfsFeatureLayer)
+        
+        // Query for features whenever map's viewpoint changes
+        self.mapView.viewpointChangedHandler = { [weak self] in
+            DispatchQueue.main.async {
+                self?.populateFeaturesFromQuery()
+            }
+        }
+        
+        // Set initial viewpoint
+        map.initialViewpoint = AGSViewpoint(targetExtent: AGSEnvelope(xMin: -122.341581, yMin: 47.613758, xMax: -122.332662, yMax: 47.617207, spatialReference: .wgs84()))
+        
+        // Assign the map to the map view
+        mapView.map = map
+
+        // Add the source code button item to the right of navigation bar
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DisplayWFSViewController"]
+    }
+    
+    private func populateFeaturesFromQuery() {
+        
+        // Create query parameters
+        let params = AGSQueryParameters()
+        params.geometry = self.mapView.visibleArea?.extent
+        params.spatialRelationship = .intersects
+        
+        // Show progress
+        SVProgressHUD.show(withStatus: "Querying")
```

Comment:
The HUD seems to rapidly disappear and appear again while interacting, rather than staying visible. Not a big deal, but I'm wondering if there's anything we can do about it? I would hope that `SVProgressHUD` is smart enough to not hide and show when asked to show when already visible. If not, there is an `isVisible` class method we could check ourselves. If we were to cancel the previous request, perhaps we could not dismiss the HUD when the error is cancelled by user? Those are just a couple ideas. I didn't try any of them.

Reply:
I think not dismissing the alert when it's a user canceled error makes more sense.